index.d.ts

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import * as PIXI from 'pixi.js';
+
+declare module 'react-pixi-fiber' {
+
+  /**
+   * The set of object keys in T not in U.
+   *
+   * Attribution: https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766:
+   */
+  export type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
+
+  /**
+   * An object with keys in T not in U.
+   *
+   * Attribution: https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766:
+   */
+  export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+
+  /** The shape of an object that has an optional `children` property of any type. */
+  interface ObjectWithChildren { children?: any; }
+
+  /** The shape of `T` without it's `children` property. */
+  export type Childrenless<T extends ObjectWithChildren> = Omit<T, 'children'>;
+
+  /** The shape of a component that has an optional `children` property. */
+  export interface ChildrenProperties {
+    children?: React.ReactNode;
+  }
+
+  /**
+   * This removes the `children` property of all component properties that inherit from `PIXI.Container`
+   * and replaces it with a `children` property appropriate for composition with JSX.
+   */
+  export type ContainerWithChildren<T extends ObjectWithChildren> = Partial<Childrenless<T>> & ChildrenProperties;
+
+  /** `BitmapText` component properties. */
+  export interface BitmapTextProperties extends ContainerWithChildren<PIXI.extras.BitmapText> {
```

Comment:
Perhaps we could define components that are not `Container`, `ParticleContainer` or `Stage` as `Childless`? Even though they extend `Container` it makes a little sense for e.g. `Sprite` to have `children`.

Reply:
Certainly.  I will define them as childless but not re-add the `children` property.

Reply:
I believe `Graphics` is intended to have children as well: http://pixijs.download/dev/docs/PIXI.Graphics.html#addChild - will prevail this.

Reply:
As you can see in the docs – this is an inherited property, all PIXI objects extending `Container` will have this, even though it doesn't make sense to use. The only way to use `Graphics` in PIXI is to draw on it, there are no `children` used.